src/gui/guiScene.cpp

```diff
@@ -0,0 +1,252 @@
+/*
+Minetest
+Copyright (C) 2020 Jean-Patrick Guerrero <jeanpatrick.guerrero@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "guiScene.h"
+
+#include <SViewFrustum.h>
+#include <IAnimatedMeshSceneNode.h>
+#include <ILightSceneNode.h>
+
+static const v2f rotation_bound{60.f, 300.f};
```

Comment:
If I get this right, you use this to make sure that the pitch is always between 0° and 60° or 300° and 360°.
- Why aren't the bounds here 90° and 270°?
- Why do you use a v2f here? Lines like `} else if (rot.X < rotation_bound.Y) {` are super confusing. Just use two variables.
- Code comments wouldn't hurt.